lib/internal/http2/core.js

```diff
@@ -307,8 +307,20 @@ function onStreamClose(code) {
 
   if (state.fd !== undefined)
     tryClose(state.fd);
-  stream.push(null);
-  stream[kMaybeDestroy](null, code);
+
+  // defer this until we actually emit end
```

Comment:
Would you be so kind and upper case the first character of each comment and punctuate the comments? :-)